### PR TITLE
Simplify tools menu creation and usage

### PR DESCRIPTION
**Pull request by makiam**

### CODE REVIEW

**Comment by Sailsman63** [comment]
Sorry for not responding earlier. My AOI time has been focused on getting 3.0.3 released. I think that this is a good refactor, however it is going to be incompatible with certain plugins. I want to get the plugin sources organized next, so that we can fix any breakage. I'm going to hold off on merging this request for the moment.

**Comment by makiam** [comment]
OK.

File: ArtOfIllusion/src/artofillusion/LayoutWindow.java [hunk 1]
```diff
@@ -549,18 +549,18 @@ public int compare(ModellingTool o1, ModellingTool o2)
...
-        item.setActionCommand("modellingTool");
-        item.addEventLink(CommandEvent.class, this, "modellingToolCommand");
-        toolsMenuItem[i] = item;
-      }
+    for (ModellingTool tool: modellingTools)
+    {
+      BMenuItem item = new BMenuItem(tool.getName());
+      item.setActionCommand("modellingTool");
+      item.addEventLink(CommandEvent.class, this, "modellingToolCommand");
+      item.getComponent().putClientProperty("tool", tool);
```
Comment (Sailsman63) [inline ArtOfIllusion/src/artofillusion/LayoutWindow.java]: Do you recall the intent of this line? I'm not sure what it is for.

**Comment by Sailsman63** [comment]
I'm revisiting this, and I don't remember why I thought that it would cause issues with plugins... perhaps it will come to me later.

File: ArtOfIllusion/src/artofillusion/LayoutWindow.java (hunk 1 above)
Comment (makiam) [reply ArtOfIllusion/src/artofillusion/LayoutWindow.java]: Line 556 sets methid to be called on this menu item invocation
Line 557 stores in menuItem which tool need to be invoked in modellingToolCommand

**Comment by Sailsman63** [comment]
Looks good! I did to a rebase, and forced update, to resolve some merge conflicts.

**Pull request merged**